libaegisub/common/vfr.cpp

```diff
@@ -225,7 +225,7 @@ int Framerate::FrameAtTime(int ms, Time type) const {
 		return int((ms * numerator / denominator - 999) / 1000);
 
 	if (ms > timecodes.back())
-		return int((ms * numerator - last + denominator - 1) / denominator / 1000) + (int)timecodes.size() - 1;
```

Comment:
Ah, thank you for fixing this! It has indeed been broken.

Reply:
Actually, after revisiting this, I think that my previous "fixed" formula also wasn't entirely correct (or at least I no longer understand why it should be correct. I didn't actually find a case where it breaks). I redid the math now to find the proper formula and explained the reasoning in the commit message. The new formula also passes all tests.